NewHorizons/Utility/Files/ImageUtilities.cs

```diff
@@ -13,41 +14,39 @@ namespace NewHorizons.Utility.Files
 {
     public static class ImageUtilities
     {
-        private static readonly Dictionary<string, Texture2D> _loadedTextures = new();
-        private static readonly List<Texture> _generatedTextures = new();
-
-        /// <summary>
-        /// Track textures generated outside of this file so they can be cleaned up on scene unload
-        /// </summary>
-        /// <param name="texture"></param>
-        public static void TrackGeneratedTexture(Texture texture) => _generatedTextures.Add(texture);
+        // key is path + applied effects
+        private static readonly Dictionary<string, Texture> _textureCache = new();
+        public static bool CheckCachedTexture(string key, out Texture existingTexture) => _textureCache.TryGetValue(key, out existingTexture);
+        public static void TrackCachedTexture(string key, Texture texture) => _textureCache.Add(key, texture);
 
         public static bool IsTextureLoaded(IModBehaviour mod, string filename)
         {
             var path = Path.Combine(mod.ModHelper.Manifest.ModFolderPath, filename);
-            return _loadedTextures.ContainsKey(path);
+            var key = path.Substring(Main.Instance.ModHelper.OwmlConfig.ModsPath.Length);
```

Comment:
We should make a method that creates these keys so we don't have to repeat this whole substring thing.